lite2/verifier.go

```diff
@@ -54,20 +54,25 @@ func Verify(
 				h1NextVals.Hash(),
 				h2.ValidatorsHash,
 			)
-			return ErrNewValSetCantBeTrusted{err}
+			return err
 		}
 	} else {
 		// Ensure that +`trustLevel` (default 1/3) or more of last trusted validators signed correctly.
 		err := h1NextVals.VerifyCommitTrusting(chainID, h2.Commit.BlockID, h2.Height, h2.Commit, trustLevel)
 		if err != nil {
-			return ErrNewValSetCantBeTrusted{err}
+			switch e := err.(type) {
+			case types.ErrNotEnoughVotingPowerSigned:
+				return ErrNewValSetCantBeTrusted{e}
```

Comment:
I might be missing something but I don't understand why not just return err, why do we need translation for this case? Couldn't `VerifyCommitTrusting()` return `ErrNewValSetCantBeTrusted` if `trustLevel` was not achieved?

Reply:
> Couldn't VerifyCommitTrusting() return ErrNewValSetCantBeTrusted if trustLevel was not achieved?

Nope because it does not know that we're checking new validator set. It only knows that vals presence in commit is some number.